test/Makefile

```diff
@@ -19,8 +19,6 @@ EMBEDDING_ARGS := "JULIA=$(JULIA_EXECUTABLE)" "BIN=$(SRCDIR)/embedding" "CC=$(CC
 
 GCEXT_ARGS := "JULIA=$(JULIA_EXECUTABLE)" "BIN=$(SRCDIR)/gcext" "CC=$(CC)"
 
-default: all
-
```

Comment:
Why?

Reply:
If I understand correctly, without this change, if you run `make test`, it will always be equivalent to running `make test all`, right? Which, as far as I can tell, is equivalent to running `make testall`. Since both of those result in running `Base.runtests("all")`,

Instead, what we want is for `make test` to be equivalent to running `make test default`, which will result in running `Base.runtests("default")`.

Reply:
With this change `make test` will be equivalent to running `make -C test all`, which is not any different. You must list the option you want explicitly.
```
diff --git a/test/Makefile b/test/Makefile
index 3d16f88a74..24e137a5b1 100644
--- a/test/Makefile
+++ b/test/Makefile
@@ -7,7 +7,7 @@ STDLIBDIR := $(build_datarootdir)/julia/stdlib/$(VERSDIR)
 # TODO: this Makefile ignores BUILDDIR, except for computing JULIA_EXECUTABLE
 
 TESTGROUPS = unicode strings compiler
-TESTS = all stdlib $(TESTGROUPS) \
+TESTS = all default stdlib $(TESTGROUPS) \
                $(patsubst $(STDLIBDIR)/%/,%,$(dir $(wildcard $(STDLIBDIR)/*/.))) \
                $(filter-out runtests testdefs, \
                        $(patsubst $(SRCDIR)/%.jl,%,$(wildcard $(SRCDIR)/*.jl))) \
@@ -19,7 +19,7 @@ EMBEDDING_ARGS := "JULIA=$(JULIA_EXECUTABLE)" "BIN=$(SRCDIR)/embedding" "CC=$(CC
 
 GCEXT_ARGS := "JULIA=$(JULIA_EXECUTABLE)" "BIN=$(SRCDIR)/gcext" "CC=$(CC)"
 
-default: all
+default:
 
 $(TESTS):
        @cd $(SRCDIR) && \
```

Reply:
Thanks Jameson! I've applied your patch.